src/screens/AddSiteScreen.js

```diff
@@ -60,9 +66,20 @@ class AddSiteScreen extends Component {
 
         setParams({currentUser});
 
-        this.attemptToUploadNewSiteIfNecessary({pendingUploadSites, uploadInProgress, props: this.props});
+        this._sub = this.props.navigation.addListener('didFocus', info => {
+            const thereAreSitesToUpload = this.thereAreSitesToUpload({pendingUploadSites});
+
+            if (thereAreSitesToUpload) {
+                this.props.checkCurrentConnectionInfo();
+            }
+        });
```

Comment:
We were having an issue where if a connection changed from poor -> good while the app was closed, when a user opened the app back up and the connection was 'good', the app wouldn't get an update and wouldn't try to upload. This listener is a nice helper that any time the `AddSiteScreen` is in focus,  and if there are sites waiting to be uploaded, we check the current connection strength in case it got good enough while we had the app closed.